src/Input.js

```diff
@@ -6,23 +6,36 @@ import uuid from 'node-uuid';
 
 class Input extends React.Component {
   componentDidMount() {
-    if (this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined') {
-      $(ReactDOM.findDOMNode(this.refs.inputEl)).material_select();
-    }
+    this._setupEl();
   }
-
   componentDidUpdate() {
-    if (this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined') {
-      var $el = $(ReactDOM.findDOMNode(this.refs.inputEl));
-      $el.material_select('destroy');
-      $el.material_select();
+    this._setupEl();
+  }
+
+  _setupEl() {
+    if (typeof $ !== 'undefined') {
+      var $inputEl = $(ReactDOM.findDOMNode(this.refs.inputEl));
+
+      if (this.props.type === 'select' && !this.props.browserDefault) {
+        $inputEl.material_select('destroy');
+        $inputEl.material_select();
+      }
+      else {
+        var $labelEl = $(ReactDOM.findDOMNode(this.refs.labelEl));
+        if ($inputEl.val()) {
+          $labelEl.addClass('active');
```

Comment:
Yes, that's what I suggested, bind the class name to a state and modify the state when the value / valueLink / defaultValue changes. It might seem to be less straight forward at the beginning, but that's required, or else react lose control of the updates. I am pretty sure it triggers `invariant` warning here.